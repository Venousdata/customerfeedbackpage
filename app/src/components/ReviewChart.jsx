// in this file i use chart.js to create a chart showing
// the most popular ratings for the selected product.

import { makeStyles } from "@material-ui/core";
import Chart from "chart.js";
import React, { useState, useEffect, useRef } from "react";
import { getReviewChartConfig } from "../helpers/helpers";

const useStyles = makeStyles((theme) => ({
  canvas: {
    height: "400px",
    width: "600px",
  },
}));

const ReviewChart = ({ reviews }) => {
  const [chartInstance, setChartInstance] = useState(null);
  const chartContainer = useRef(null);
  const classes = useStyles();

  const chartConfig = getReviewChartConfig(reviews);

  useEffect(() => {
    // basically the react hooks way of doing document.getElementById
    // the element (canvas) is then passed to chart.js to render the chart
    if (chartContainer && chartContainer.current) {
      const newChartInstance = new Chart(chartContainer.current, chartConfig);
      setChartInstance(newChartInstance);
    }
  }, [chartContainer, reviews]);

  return (
    <div>
      <canvas ref={chartContainer} className={classes.canvas} />
    </div>
  );
};

export default ReviewChart;
