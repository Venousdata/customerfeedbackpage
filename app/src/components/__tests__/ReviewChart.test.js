import React from "react";
import { render, screen } from "@testing-library/react";
import ReviewChart from "../ReviewChart";
import { BrowserRouter } from "react-router-dom";
import { mockReviews } from "./mockData/ReviewChart.data";

describe("ReviewChart tests", () => {
  it("ReviewChart should render and match snapshot", (done) => {
    // hack to get chart.js to play nicely with @testing-library/react
    jest.mock("chart.js", () => "Chart");

    render(
      <BrowserRouter>
        <ReviewChart reviews={mockReviews} />
      </BrowserRouter>
    );

    const component = screen.getByTestId("reviewchart");
    expect(component.innerHTML).toMatchSnapshot();
    done();
  });
});
