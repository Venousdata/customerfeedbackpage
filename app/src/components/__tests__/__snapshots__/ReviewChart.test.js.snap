// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewChart tests ReviewChart should render and match snapshot 1`] = `"<canvas class=\\"makeStyles-canvas-1\\"></canvas>"`;
